### PR TITLE
Write sbom.legacy.json files for newer platform api

### PR DESCRIPTION
**Pull request by natalieparellano**
Resolves https://github.com/buildpacks/lifecycle/issues/808

### CODE REVIEW

File: utils.go [hunk 1]
```diff
@@ -1,41 +1,9 @@
...
-	return group, err
-}
-
-func ReadOrder(path string) (buildpack.Order, error) {
-	var order struct {
-		Order buildpack.Order `toml:"order"`
-	}
-	_, err := toml.DecodeFile(path, &order)
-	return order.Order, err
-}
```
Comment (natalieparellano) [inline utils.go:37]: These functions are unused. I think we forgot to delete them in an earlier PR.

File: builder.go [hunk 2]
```diff
@@ -120,12 +123,23 @@ func (b *Builder) Build() (*platform.BuildMetadata, error) {
 		}
 	}
 
+	if b.Platform.API().AtLeast("0.9") {
+		b.Logger.Debug("Creating sBOM files for legacy BOM")
+		if err := encoding.WriteJSON(filepath.Join(b.LayersDir, "sbom", "launch", "sbom.legacy.json"), launchBOM); err != nil {
+			return nil, errors.Wrap(err, "encoding launch bom")
+		}
+		launchBOM = []buildpack.BOMEntry{} // zero out bom in <layers>/config/metadata.toml
```
[the remark this answers is not in the document]
Comment (natalieparellano) [reply builder.go]: Fixed in db5d4af2 - thanks @jabrown85 for the suggestion!

File: internal/encoding/utils.go [hunk 3]
```diff
@@ -1,12 +1,35 @@
...
 import (
+	"bytes"
+	"encoding/json"
 	"os"
 	"path/filepath"
 
 	"github.com/BurntSushi/toml"
 )
 
+func MarshalJSON(v interface{}) ([]byte, error) { // TODO: test
```
Comment (jabrown85) [inline internal/encoding/utils.go:12]:
```suggestion
func MarshalTOML(v interface{}) ([]byte, error) { // TODO: test
```

**Review by mboldt — APPROVED** [review]
Added a couple suggestions, but overall looks good to me!

File: builder.go [hunk 4]
```diff
@@ -120,12 +123,23 @@ func (b *Builder) Build() (*platform.BuildMetadata, error) {
 		}
 	}
 
+	if b.Platform.API().AtLeast("0.9") {
+		b.Logger.Debug("Creating sBOM files for legacy BOM")
```
[the remark this answers is not in the document]
Comment (natalieparellano) [reply builder.go]: Added in d09d8f04

**Review by jabrown85 — APPROVED** [review]

**Pull request merged**